test/Modules/UngradedModules.jl

```diff
@@ -0,0 +1,663 @@
+using Random
+
+RNG = Random.MersenneTwister(42)
+
+"""
+	randpoly(R::Ring,coeffs=0:9,max_exp=4,max_terms=8)
+> Return a random Polynomial from the Polynomial Ring `R` with coefficients in `coeffs`
+> with exponents between `0` and `max_exp` und between `0` and `max_terms` terms
+"""
+function randpoly(R::Oscar.Ring,coeffs=0:9,max_exp=4,max_terms=8)
+	n = nvars(R)
+	K = base_ring(R)
+	E = [[Random.rand(RNG,0:max_exp) for i=1:n] for j=1:max_terms]
+	C = [K(Random.rand(RNG,coeffs)) for i=1:max_terms]
+	M = MPolyBuildCtx(R)
+	for i=1:max_terms
+		push_term!(M,C[i],E[i])
+	end
+	return finish(M)
+end
+
+"""
+	matrix(A::Array,R::Ring = parent(A[1,1]))
+Return `A` as an AbstractAlgebra Matrix
+"""
+function Oscar.matrix(A::Array,R::AbstractAlgebra.Ring = parent(A[1,1]))
```

Comment:
Huh? Normally it's `matrix(R, A)` (other argument order), and already exists. Same for `matrix(A)` where the ring can be derived from the matrix `A`.

What does this method do that `matrix(R, A)` (resp. `matrix(R, n, m, A)` doesn't do? Even if there is something, the different argument order is bad.